libjob_queue/src/job_queue.c

```diff
@@ -765,17 +754,12 @@ static void job_queue_handle_DO_KILL_NODE_FAILURE(job_queue_type * queue, job_qu
 
 static void job_queue_handle_DO_KILL( job_queue_type * queue , job_queue_node_type * node) {
   job_queue_kill_job_node(queue, node);
-  job_queue_change_node_status(queue , node , JOB_QUEUE_RUNNING_CALLBACK );
-  {
-    arg_pack_type * arg_pack = arg_pack_alloc();
-    arg_pack_append_ptr( arg_pack , queue );
-    arg_pack_append_int( arg_pack , job_queue_node_get_queue_index(node));
-    thread_pool_add_job( queue->work_pool , job_queue_run_DO_KILL_callback , arg_pack );
-  }
+  job_queue_node_free_driver_data( node , queue->driver );
+  job_queue_change_node_status(queue , node , JOB_QUEUE_IS_KILLED );
```

Comment:
I understand why the kill callback shouldn't be called on node failure, but I don't get why we shouldn't call it when the user excplicitly ask to kill the job (but I have a feeling you are considering to remove the kill callback completely...)

Reply:
The state `JOB_QUEUE_RUNNING_KILL_CALLBACK` was removed during this PR. The point is that killing a job is done syncronously, so then it is not necessary to (temporarily) go into a callback state.